CONTRIBUTING.md

```diff
@@ -738,6 +738,45 @@ meta = dict['metadata']
 myrank = meta['mpi_rank']
 ```
 
+### Running with Threadpool and DASK
```

Comment:
We should move this and the MPI docs in contributing to a page in the API docs, this can be done as a follow up though